substrate/frame/contracts/src/benchmarking/mod.rs

```diff
@@ -152,6 +156,34 @@ where
 		Ok(())
 	}
 
+	/// Create a new contract with the specified unbalanced storage trie.
+	fn with_unbalanced_storage_trie(
+		code: WasmModule<T>,
+		trie_layers: u32,
+		key_size: u32,
+		value_size: u32,
+	) -> Result<Self, &'static str> {
+		if key_size < (trie_layers + 1) / 2 {
+			return Err("Key size to small to create the specified trie");
+		}
+		let contract = Contract::<T>::new(code, vec![])?;
+		let info = contract.info()?;
+		let child_trie_info = info.child_trie_info();
+		for l in 0..trie_layers {
+			let pos = l as usize / 2;
+			let mut key = vec![0u8; key_size as usize];
+			for i in 0u8..16 {
+				if l % 2 == 0 {
+					key[pos] = i
+				} else {
+					key[pos] = i << 4;
+				}
```

Comment:
```suggestion
				key[pos] = if l % 2 == 0 { i } else { i << 4 };
```

Reply:
don't you overwrite your keys here since every other iteration will produce the same set of keys 🤔

Reply:
0 key should be overwritten only